### PR TITLE
winPB: automate jenkins service installation

### PR DESCRIPTION
**Pull request by gdams**
##### Checklist

- [x] commit message has one of the [standard prefixes](https://github.com/adoptium/infrastructure/blob/master/CONTRIBUTING.md#commit-messages)
- [ ] [faq.md](https://github.com/adoptium/infrastructure/blob/master/FAQ.md) updated if appropriate
- [x] other documentation is changed or added (if applicable)
- [ ] playbook changes run through [VPC](https://ci.adoptium.net/view/Tooling/job/VagrantPlaybookCheck/) or [QPC](https://ci.adoptium.net/view/Tooling/job/QEMUPlaybookCheck/) (if you have access)
- [ ] VPC/QPC not applicable for this PR
- [x] for inventory.yml changes, bastillion/nagios/jenkins updated accordingly

### CODE REVIEW

File: ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/Jenkins_Service_Installation/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    path: C:\Users\{{ Jenkins_Username }}\agent
+    state: directory
+    recurse: yes
+    owner: "{{ Jenkins_Username }}"
+  when: jenkins_service.exists == false
+
+# TODO: Add support for Windows Aarch64 native binary once it's available
+- name: Download WIN-SW
+  win_get_url:
+    url: https://github.com/winsw/winsw/releases/download/v2.12.0/WinSW-x64.exe
```
Comment (karianna) [inline ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/Jenkins_Service_Installation/tasks/main.yml:37]: Do we need to verify that download?

File: ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/Jenkins_Service_Installation/templates/JenkinsAgentService.xml.j2 [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<service>
+	<id>Jenkins</id>
+	<name>Jenkins</name>
+	<description>This service runs an agent for Jenkins automation server.</description>
+	<executable>C:\openjdk\jdk-17\bin\java.exe</executable>
```
Comment (karianna) [inline ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/Jenkins_Service_Installation/templates/JenkinsAgentService.xml.j2:5]: extract version?

File: ansible/playbooks/AdoptOpenJDK_Windows_Playbook/main.yml [hunk 3]
```diff
@@ -83,6 +83,9 @@
     - NSClient                    # Required For Nagios Monitoring
     - shortNames
     - Dragonwell                  # Dragonwell bootstrap image
+    - role: Jenkins_Service_Installation # Automate installing the jenkins service
+      tags: [jenkins, adoptopenjdk]
```
[the remark this answers is not in the document]
Comment (gdams) [reply ansible/playbooks/AdoptOpenJDK_Windows_Playbook/main.yml:87]: we likely should but that would be a change to many existing roles. I'd suggest doing this in a separate PR

File: ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/Jenkins_Service_Installation/tasks/main.yml (hunk 1 above)
Comment (gdams) [reply ansible/playbooks/AdoptOpenJDK_Windows_Playbook/roles/Jenkins_Service_Installation/tasks/main.yml:37]: yup

**Review by sxa — CHANGES_REQUESTED** [review]
Can we get a doc update along with this (e.g. to https://github.com/adoptium/infrastructure/tree/master/ansible/playbooks/AdoptOpenJDK_Windows_Playbook/README.md) to cover the variable overrides that may be required (e.g. secret and workspace dir)

(Noting that such an update will conflict with https://github.com/adoptium/infrastructure/pull/2993/files but if it's a separate paragraph it'll be easy to merge :-) )

**Review by karianna — APPROVED** [review]

**Comment by gdams** [comment]
> Can we get a doc update along with this (e.g. to

@sxa I've added some docs

**Comment by karianna** [comment]
Playbook failures related?

**Comment by gdams** [comment]
> Playbook failures related?

Unrelated

**Review by sxa — APPROVED** [review]
LGTM - thanks for the doc updates

**Pull request merged**